src/modulation/gfsk.rs

```diff
@@ -0,0 +1,15 @@
+//! Common GFSK modulation options
+
+/// Basic GFSK channel configuration
+#[derive(Clone, PartialEq, Debug)]
+#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
+pub struct GfskChannel {
+    /// Channel frequency in kHz
```

Comment:
Can't we make this Hz? It would top out at 4.2GHz, but is that going to limit any use case we have in mind?

Reply:
the decawave gear (DWM1000 etc.) goes up to 6.5GHz so, i think we're better to keep it wider. do you see a need for sub-khz definitions? because if so we could have a more complex `Frequency` type

Reply:
(and ngl i would love to have a proper parser that understood frequency units)

Reply:
I don't have a concrete use case, but this feels like a requirement which is not going to be true for all implementations.

What about using embedded time Rate with concrete type being an associated type which defaults to kHz?

https://docs.rs/embedded-time/0.12.1/embedded_time/rate/trait.Rate.html

Reply:
at the moment we're avoiding `embedded_time` in `e-h` due to the complexity, so, i think i'd prefer a simpler type defined by us at the moment.

Reply:
Makes sense, we can define our own Frequency trait and implement that for e-t Rate (behind a feature). Then users can decide whether they bring their own type or use e-t.

RTIC uses a similar approach for their Monotonic: https://github.com/rtic-rs/rtic-monotonic/blob/4050b236f36ef99da2d88a9e2d4a211a8d46d262/src/lib.rs#L34